dev/PushNotifications/PushNotificationsLongRunningTask/NotificationListener.cpp

```diff
@@ -32,7 +37,19 @@ STDMETHODIMP_(HRESULT __stdcall) NotificationListener::OnRawNotificationReceived
 
     if (!m_foregroundSinkManager->InvokeForegroundHandlers(m_appId, payloadArray, payloadLength))
     {
-        THROW_IF_FAILED(winrt::Microsoft::Windows::PushNotifications::Helpers::ProtocolLaunchHelper(m_processName, payloadLength, payload));
+        if (m_comServerClsid == winrt::guid())
+        {
+            THROW_IF_FAILED(winrt::Microsoft::Windows::PushNotifications::Helpers::ProtocolLaunchHelper(m_processName, payloadLength, payload));
+        }
+        else
+        {
+            std::wstring payloadString(payloadArray.begin(), payloadArray.end());
+
+            auto localBackgroundTask = winrt::create_instance<winrt::Windows::ApplicationModel::Background::IBackgroundTask>(m_comServerClsid, CLSCTX_ALL);
+            auto pushBackgroundTaskInstance{ winrt::make_self<PushBackgroundTaskInstance>() };
+            pushBackgroundTaskInstance->SetRawNotificationPayload(payloadString);
+            localBackgroundTask.Run(*pushBackgroundTaskInstance);
```

Comment:
Run returns HRESULT. So THROW_IF_FAILED here

Reply:
PushNotificationBackgroundTask::Run is a void function